File: credit.c
#include <stdio.h>
#include <cs50.h>


bool isValidCard( long long passedInNumber);
void printType(long long number);

int main(void)
{
    printf("Number: ");
    long long cnumber = get_long_long();
    printType(cnumber);
} 


bool isValidCard( long long passedInNumber) {
    long long number = passedInNumber;
    bool isOdd = true;
    int sum = 0;
    while (number > 0) 
    {
        int digit = number % 10;
        if (isOdd) 
        {
            sum += digit;
        } 
        else 
        {
            int tmp = digit * 2;
            if (tmp > 9) 
            {
                tmp = 1 + tmp%10;
            }
            sum += tmp;
        }
        number  = number / 10;
        isOdd = !isOdd;
    }
    return (sum%10 == 0);
}


void printType(long long number) {
    if (isValidCard(number)) 
    {
        while(number>100)
        {
            number=number/10;
        }
        if(number == 34 || number == 37)
        {
            printf("AMEX\n");
        }
        else if(number >= 51 && number <= 55)
        {
            printf("MASTERCARD\n");
        }
        else if(number >= 40 && number <= 49)
        {
            printf("VISA\n");
        }
        else printf("INVALIDCARDTYPE\n");
    }
    else
    {
        printf("%s\n", "INVALID");
    }
}

File: credit.py
import cs50

def main():
    print("Number: ", end="")    
    cardNumber = cs50.get_float()
    printType(cardNumber)
    
def isValidCard(cardNumber):
    cNumber = cardNumber
    isOdd = True
    sum = 0
    while cNumber > 0:
        digit = cNumber % 10
        if isOdd:
            sum += digit;
        else:
            temp = digit*2
            if temp > 9:
                temp = 1 + temp%10
            sum += temp
        cNumber = cNumber // 10
        isOdd = not(isOdd)
    return (sum%10 == 0)
    
def printType(cardNumber):
    if isValidCard(cardNumber):
        while cardNumber > 100:
            cardNumber = cardNumber // 10
        if cardNumber == 34 or cardNumber == 37:
            print("AMEX")
        elif cardNumber >= 51 and cardNumber <= 55:
            print("MASTERCARD")
        elif cardNumber >= 40 and cardNumber <= 49:
            print("VISA")
        else:
            print("INVALIDCARDTYPE")
    else:
        print("INVALID")


if __name__ == "__main__":
    main()
